### ISSUE TITLE
[Due for payment 2025-03-10] [$500] Android - Chat - Code block alignment is too high in Android

### ISSUE THREAD

**Issue by vincdargento** [Expensify/App#53458]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.70-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** N/A
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Launch app in both mweb and android
2. Open a chat
3. Enter Testing "`hello`" testing
4. Send the message

## Expected Result:
Code blocks must display consistently in both mweb and android.

## Actual Result:
In mWeb, code block alignment is correct, while in Android, code block alignment is too high.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/b383553f-a650-4cff-999e-55711c4d0a18

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021864441841607834076</li>
        <li>Upwork Job ID: 1864441841607834076</li>
        <li>Last Price Increase: 2025-03-04</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sakluger</details>

**Comment by sakluger** [comment]
I think that the quotation mark alignment is the same for both platforms, while the code snippet alignment is different. In Android, the code snippet looks much higher and closer to the text above than in mWeb. @Expensify/design could I get a gut check on that?

Pulling out screenshots from the video, since it took me a while to see the different while watching the video.

Here's mWeb:

<img width="680" alt="image" src="https://github.com/user-attachments/assets/8919b336-95b8-4567-b475-fe4b02936277">

Here's Android:

<img width="547" alt="image" src="https://github.com/user-attachments/assets/f5ed71d2-c6ca-418e-b603-924fb7f8d9f6">

**Comment by dubielzyk-expensify** [comment]
Agree that the code snipped looks a tad high, @sakluger . 

Also, could we see how it looks when you insert more text like this:
![CleanShot 2024-12-05 at 08 22 32@2x](https://github.com/user-attachments/assets/48f49bba-2c74-44be-8d2b-eddc58511537)

Cause that'll give us a better idea of what's actually misaligned here

**Comment by sakluger** [comment]
Good suggestion! Here's that same longer string in Android - it's pretty clear that the code block is aligned super high:

<img width="298" alt="image" src="https://github.com/user-attachments/assets/e807ab92-a7fb-482f-b068-d1eb1d89757e">

**Comment by sakluger** [comment]
I updated the GH issue title and the actual/expected behavior in the OP.

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The quoted test is showing smaller, space in the quoted message is interrupted, and not showing in the same line (Higher than the rest of the text)

### What is the root cause of that problem?

The `lineHeight` of normal text is `20`:

https://github.com/Expensify/App/blob/08e571ca195023f63160076d560f92de96d0fe49/src/styles/index.ts#L240

https://github.com/Expensify/App/blob/2c561aee7b9e92f086a02b088c9f66934d93ca93/src/styles/variables.ts#L66

while the `lineHeight` of the code text and `height` of the code wrapper on Android is much smaller compared to `20`:

https://github.com/Expensify/App/blob/82b76c1a36cf1a01b00c14510a5812a591e7f2fd/src/styles/utils/codeStyles/index.android.ts#L3-L14

### What changes do you think we should make in order to solve the problem?


My experiment proved that we can reuse [iOS's `codeStyles`](https://github.com/Expensify/App/blob/82b76c1a36cf1a01b00c14510a5812a591e7f2fd/src/styles/utils/codeStyles/index.ios.ts) for Android. So we can remove the Android specific file and rename iOS file to `.native.ts` to apply to both platforms.

**Comment by sakluger** [comment]
@getusha could you please review the above proposal when you have a chance?

**Comment by sakluger** [comment]
@allgandalf could you please review the proposal above when you have a chance?

**Comment by mkzie2** [comment]
Let's just adjust the `codeWordWrapper`'s `height` and the `codeTextStyle`'s `lineHeight` in Android to match iOS for safety.

**Comment by mkzie2** [comment]
@allgandalf Thanks. I updated.

**Comment by sakluger** [comment]
Georgia may be out for the holidays, along with most of the team. I'm going to set to weekly for now.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by grgia** [comment]
@allgandalf can you confirm your choice after considering @parasharrajat's [comment](https://github.com/Expensify/App/issues/53458#issuecomment-2566460136)

**Comment by grgia** [comment]
Moving back to daily

**Comment by shawnborton** [comment]
Sounds like this is happening on iOS too, not just Android?

**Comment by JmillsExpensify** [comment]
I just hit this one this morning on iOS, so yes.

**Comment by sakluger** [comment]
We usually don't pay more than $250 for alignment issues (in fact alignment issues usually have a lower bounty). @JmillsExpensify @grgia how do you feel about bumping the bounty here?

**Comment by sakluger** [comment]
Asked about prioritization in Slack: https://expensify.slack.com/archives/C01SKUP7QR0/p1737073037381949

**Comment by sakluger** [comment]
No response on the Slack post. I'll ask again tomorrow.

**Comment by sakluger** [comment]
Bumped the Slack post.

**Comment by sakluger** [comment]
In the Slack thread we decided to close this issue. It’s not worth increasing the bounty, so if it’s too complex to fix for the current amount, then we should just close it.

**Comment by Beamanator** [comment]
FYI I just reported this issue in slack here - https://expensify.slack.com/archives/C05LX9D6E07/p1737728009186649

It's looking reeeeeal bad in iOS & still floats high on Android. In my report I noted more examples of reproducing, I think we should reopen

**Comment by sakluger** [comment]
We've decided to reopen this GH issue and increase the bounty to encourage more proposals.

**Comment by hungvu193** [comment]
I think this one is a dupe of https://github.com/Expensify/App/issues/49935. We decided to close it before.

Here's the RCA: https://github.com/Expensify/App/issues/8132#issuecomment-1436734572

**Comment by sakluger** [comment]
@allgandalf I'd like to wait for some proposals from Contributors at the new bounty before posting to SWM and CS channels.

**Comment by grgia** [comment]
@sakluger @allgandalf any updates on this one?

**Comment by sakluger** [comment]
I bumped @allgandalf on Slack: https://expensify.slack.com/archives/C02NK2DQWUX/p1738780860736239

**Comment by grgia** [comment]
Thank you!

**Comment by grgia** [comment]
Why melvin

**Comment by fabioh8010** [comment]
@allgandalf 

Yes this is indeed a very old issue, since 2022 ([here](https://github.com/Expensify/App/issues/8132)). The root problem is on native side as explained [here](https://github.com/Expensify/App/issues/8132#issuecomment-1436734572), and [here's the RN issue](https://github.com/facebook/react-native/issues/31955). I proposed to proceed with a [workaround to improve alignment (but not fix it)](https://github.com/Expensify/App/pull/15367) while I was going to investigate and work in a proper fix for react-native.

At that time [I suggested one of these approaches](https://github.com/Expensify/App/issues/8132#issuecomment-1616011282), but the react-native PRs ended up being closed by Meta's decision, so I would say we currently have these options:

1. Proceed with a **temporary** solution to improve alignment like [this one](https://github.com/Expensify/App/pull/15367) but since it's very old we need to confirm if it still works.
2. Based on [this discussion on Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1677360137423139?thread_ts=1676991093.760849&cid=C01GTK53T8Q), simplify the code block styling in a way that won't cause the issue anymore, but it's a design change and needs approval from the design team.
3. Explore other solutions on native side [like the one I suggested](https://github.com/Expensify/App/issues/8132#issuecomment-1436734572).

**Comment by sakluger** [comment]
@Expensify/design thoughts on @fabioh8010's suggested solution 2 above?

> Based on [this discussion on Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1677360137423139?thread_ts=1676991093.760849&cid=C01GTK53T8Q), simplify the code block styling in a way that won't cause the issue anymore, but it's a design change and needs approval from the design team.

**Comment by shawnborton** [comment]
@fabioh8010 can you show us screenshots of what 2 would look like in practice? And of the options you listed out, is option 1 okay to pursue? Do you have a recommendation for which of the options you think is best?

**Comment by dangrous** [comment]
hi all, I just closed https://github.com/Expensify/App/issues/56207 because it seems like the same bug - but on iOS. I'm guessing it'll have the same solution? If not please let me know and I can reopen.

**Comment by fabioh8010** [comment]
> hi all, I just closed [#56207](https://github.com/Expensify/App/issues/56207) because it seems like the same bug - but on iOS. I'm guessing it'll have the same solution? If not please let me know and I can reopen.

Yes @dangrous, solution applies to both platforms!

---

@shawnborton Here's a [POC PR](https://github.com/Expensify/App/pull/56924) for `Option 2 (simplify the code block styling)`. You can trigger a test build in case you want to test by yourself, but I will attach some screenshots here as well.

<details>
<summary>Dark theme</summary>

| Before | After | After (more contrast) |
| --- | --- | --- |
| ![Image](https://github.com/user-attachments/assets/5207acb9-89fc-453a-b6e3-44b342a53142) | ![Image](https://github.com/user-attachments/assets/102ba275-6222-4ee2-aef8-87bff06a725d) | ![Image](https://github.com/user-attachments/assets/234f4d8f-1ac9-4643-b127-4cfa203d7388) |
| ![Image](https://github.com/user-attachments/assets/06bef5cc-91c3-4a5a-a17d-2bfc48cb382b) | ![Image](https://github.com/user-attachments/assets/2449176f-cd82-4089-ac3c-67923b25c0de) | ![Image](https://github.com/user-attachments/assets/47a74cda-fdb8-4c2f-b84a-4e337616105d) |

</details>

<details>
<summary>Light theme</summary>

| After | After (more contrast) |
| --- | --- |
| ![Image](https://github.com/user-attachments/assets/31944779-b08a-48bd-a00c-afbdfaaa7032) | ![Image](https://github.com/user-attachments/assets/c5ff0c0d-69a2-42fe-876e-7fad5b82450a) |
| ![Image](https://github.com/user-attachments/assets/2dc25d83-7a4e-4dd7-92f9-611274ffed24) | ![Image](https://github.com/user-attachments/assets/afc351ae-bc7f-4d4c-ad16-9df2148ba2e8) |

</details>

**Comment by shawnborton** [comment]
Awesome, thanks - we can continue to work in the PR. I think the `After` version (with no constrast adjustments) is probably fine to start? 

And just to clarify, outside of doing this option, you are going to pursue Option 3 so we can fix this the real way hopefully once and for all right?

**Comment by fabioh8010** [comment]
> Awesome, thanks - we can continue to work in the PR. I think the `After` version (with no constrast adjustments) is probably fine to start?
> 
> And just to clarify, outside of doing this option, you are going to pursue Option 3 so we can fix this the real way hopefully once and for all right?

Yes, after this PR is merged we can create a follow-up issue to explore the native fix for it. Just wanted to highlight that it will certainly require some effort to get this done as explained before, but I assume it's worth the value, correct?

**Comment by shawnborton** [comment]
I think it's worth it, yup. I feel like we really need to tame all of these various text styles we have across the platforms.

**Comment by fabioh8010** [comment]
> I think it's worth it, yup. I feel like we really need to tame all of these various text styles we have across the platforms.

Alright! 👍 

**Updates:**

- [PR](https://github.com/Expensify/App/pull/56924) is ready for complete review.

**Comment by fabioh8010** [comment]
**Updates:**

- [PR](https://github.com/Expensify/App/pull/56924) was merged!

@shawnborton let's create a follow-up issue to address the fix on native side?

**Comment by shawnborton** [comment]
Can do! Issue is here if you don't mind commenting on it and I will assign to you: https://github.com/Expensify/App/issues/57556

**Comment by shawnborton** [comment]
Can do

**Comment by sakluger** [comment]
Hey @allgandalf and @grgia - I saw [this comment](https://github.com/Expensify/App/issues/57592#issuecomment-2709549892) mention that this issue's PR caused a follow-up issue. Would we consider that a regression, or is that follow-up work out of scope of this issue?

**Comment by sakluger** [comment]
I'm going to send the contract via Upwork now. If this issue caused a regression, we can always adjust the payout.

**Comment by sakluger** [comment]
Thanks @allgandalf! I trust you.

Summarizing payment on this issue:

Contributor: @fabioh8010 - no payment required
Contributor+: @allgandalf $500, paid via Upwork
